store/views.py

```diff
@@ -77,6 +99,41 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    response_data = {
+        'message': "Book is successfully returned! Watch for more books.",
+    }
+    book_id = request.POST.get('bid') # get the book id from post data
+    book = Book.objects.filter(pk=book_id)[0]
+    bookcopy=BookCopy.objects.filter(book=book).filter(status=False)[0] 
+    bookcopy.borrower=None
+    bookcopy.borrow_date=None
+    bookcopy.status=True
+    bookcopy.save()
+    return JsonResponse(response_data)
 
+@csrf_exempt
+@login_required
+def rateBook(request):
+    response_data = {
+        'message': "Successfully rated!",
+    }
+    username =request.user.username
+    book_id = request.POST.get('bid') # get the book id from post data
+    book = Book.objects.filter(pk=book_id)[0]
+    newrating = BookRating.objects.filter(book=book,rater=username)
+    if(len(newrating)==0):
+        response_data['message']="You have not issued the book!"
+        return JsonResponse(response_data)
+    newrating=newrating[0]
+    rate = request.POST.get('rate')
+    newrating.rated=True
+    newrating.rating=rate
+    newrating.save()
 
+    bookratings=BookRating.objects.filter(book=book,rated=True)
+    sum=0.0
+    for i in range(len(bookratings)):
+        sum=sum+bookratings[i].rating
+    book.rating=round((float)(sum/len(bookratings)), 1)
```

Comment:
This is correct but a more clean and Django-ish way for this could be (with looping):
```python
book.rating = BookRating.objects.filter(book__pk=book_id).aggregate(Avg('rating'))['rating__avg']
```

Reply:
Ok I'll make sure to use these functions from next time.